cloudferrylib/os/actions/detach_used_volumes_via_compute.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+import copy
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import utils as utl
+
+LOG = utl.get_log(__name__)
+
+
+class DetachVolumesCompute(action.Action):
+
+    def run(self, info, **kwargs):
+        info = copy.deepcopy(info)
+        compute_resource = self.cloud.resources[utl.COMPUTE_RESOURCE]
+        storage_resource = self.cloud.resources[utl.STORAGE_RESOURCE]
+        for instance in info[utl.INSTANCES_TYPE].itervalues():
+            LOG.debug("Work on instance %s [%s]" %
```

Comment:
Would be more readable with message "Detaching volumes for instance {name} ({id})", "Work on instance" has close to no value for the end-user.